src/fileio.cpp

```diff
@@ -51,85 +48,86 @@ const char* LoadingError::what() const noexcept
 }
 
 /**
- * @fn void DirectoryReader::OpenPath(const char *path)
- * Set up the directory reader object for reading a directory.
- * @param path Path to the directory.
- * @note Failure is not reported here, but #NextEntry will not return anything useful.
+ * Scan a given directory and return all entries.
+ * @param Directory to scan.
  */
+std::vector<std::string> GetAllEntries(const std::string &path)
+{
+	std::vector<std::string> entries;
+	if(!PathIsDirectory(path)) return entries;
```

Comment:
Several whitespace problems in this file: A whitespace is mandatory after the keywords `if` and `for`.